libraries/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -178,17 +196,29 @@ export class ProjectChangeAnalyzer {
     return projectState;
   }
 
+  /**
+   * Given a specific project and a list of changed files for that project, apply project-specific
+   * "ignore globs" settings, as well as an optional filtering function, returning a new list of
+   * changed files that have passed through the filter.
+   */
   public async _filterProjectDataAsync<T>(
     project: RushConfigurationProject,
     unfilteredProjectData: Map<string, T>,
     rootDir: string,
-    terminal: ITerminal
+    terminal: ITerminal,
+    filterByProject?: ProjectFilterFactory
   ): Promise<Map<string, T>> {
-    const ignoreMatcher: Ignore | undefined = await this._getIgnoreMatcherForProjectAsync(project, terminal);
-    if (!ignoreMatcher) {
+    let ignoreMatcher: Ignore | undefined = await this._getIgnoreMatcherForProjectAsync(project, terminal);
+    if (!ignoreMatcher && !filterByProject) {
       return unfilteredProjectData;
     }
+    if (!ignoreMatcher) {
+      ignoreMatcher = ignore();
+    }
```

Comment:
I seem to recall this call being somewhat expensive, which is why `_getIgnoreMatcherForProjectAsync` returns `undefined` in the first place.